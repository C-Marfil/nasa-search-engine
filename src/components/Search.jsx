import React from "react";
import "../styles/search.css";

const Search = () => {
  return (
    <>
      <input className="search-box" type="text" placeholder="Search here" />
      <button className="button" type="submit">
        GO
      </button>
    </>
  );
};

export default Search;
