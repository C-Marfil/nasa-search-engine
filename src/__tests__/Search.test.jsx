import { React, render } from "@testing-library/react";
import Search from "../components/Search";

describe("Search", () => {
  it("Snapshot Test", () => {
    const { asFragment } = render(<Search />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("Checks that elements render properly", () => {
    const { getByText, getByRole } = render(<Search />);

    expect(getByText("GO")).toBeInstanceOf(HTMLButtonElement);
    expect(getByRole("textbox")).toHaveAttribute("class", "search-box");
  });
});
