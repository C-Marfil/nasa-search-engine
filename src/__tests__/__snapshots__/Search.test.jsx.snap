// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Snapshot Test 1`] = `
<DocumentFragment>
  <input
    class="search-box"
    placeholder="Search here"
    type="text"
  />
  <button
    class="button"
    type="submit"
  >
    GO
  </button>
</DocumentFragment>
`;
